### PR TITLE
Improve CreateFakeJwks

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/istio/istio/pull/47532

see details: 

https://github.com/istio/istio/pull/47532#issuecomment-1778855094
https://github.com/istio/istio/pull/47532#issuecomment-1778882015

### CODE REVIEW

File: pilot/pkg/model/jwks_resolver.go [hunk 1]
```diff
@@ -115,6 +117,17 @@ type jwtKey struct {
 	issuer  string
 }
 
+type CreateFakeJwksFunc func() string
+
+// CreateFakeJwks is a helper function to make a fake jwks when istiod failed to fetch it.
+func CreateFakeJwks() string {
```
Comment (howardjohn) [inline pilot/pkg/model/jwks_resolver.go:123]: Why does it need to be dynamic? This is highly costly in the control plane (RSA key is very very expensive) and will change on each xDS (costly in envoy to reload, and drain connections)?

Comment (zirain) [reply pilot/pkg/model/jwks_resolver.go]: make sense, update.

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/pkg/model/jwks_resolver.go [hunk 2]
```diff
@@ -296,12 +299,11 @@ func (r *JwksResolver) BuildLocalJwks(jwksURI, jwtIssuer, jwtPubKey string) *env
 }
 
 // CreateFakeJwks is a helper function to make a fake jwks when istiod failed to fetch it.
-func CreateFakeJwks(jwksURI string) string {
-	// Create a fake jwksURI
-	fakeJwksURI := "Error-IstiodFailedToFetchJwksUri-" + jwksURI
-	// Encode jwksURI with base64 to make dynamic n in jwks
-	encodedString := base64.RawURLEncoding.EncodeToString([]byte(fakeJwksURI))
-	return fmt.Sprintf(`{"keys":[ {"e":"AQAB","kid":"abc","kty":"RSA","n":"%s"}]}`, encodedString)
+func CreateFakeJwks() string {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/jwks_resolver.go:302]: Is this a const, can we cache it

Comment (zirain) [reply pilot/pkg/model/jwks_resolver.go:302]: updated

Comment (howardjohn) [reply pilot/pkg/model/jwks_resolver.go:302]: This is still not stable and will result in XDS draining as we hop between instances every 30min. This is really bad and this exact issue has caused massive production issues for our customers in the past./

This must be static and deterministic.

**Pull request merged**